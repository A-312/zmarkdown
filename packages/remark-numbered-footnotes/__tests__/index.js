import {readdirSync as directory, readFileSync as file} from 'fs'
import {join} from 'path'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

const base = join(__dirname, 'fixtures')
const specs = directory(base).reduce((tests, contents) => {
  const parts = contents.split('.')
  if (!tests[parts[0]]) {
    tests[parts[0]] = {}
  }
  tests[parts[0]][parts[1]] = file(join(base, contents), 'utf-8')
  return tests
}, {})

const config = {
  gfm: true,
  commonmark: false,
  footnotes: true
}

test('footnotes', () => {
  const {contents} = unified()
    .use(reParse, config)
    .use(require('../src'))
    .use(remark2rehype)
    .use(stringify)
    .processSync(specs['footnotes'].fixture)

  expect(contents).toMatchSnapshot()
})

test('regression-1', () => {
  const {contents} = unified()
    .use(reParse, config)
    .use(require('../src'))
    .use(remark2rehype)
    .use(stringify)
    .processSync(specs['regression-1'].fixture)

  expect(contents).toMatchSnapshot()
})
