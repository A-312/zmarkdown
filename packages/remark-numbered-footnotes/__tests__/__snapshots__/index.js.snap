// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footnotes 1`] = `""`;

exports[`regression-1 1`] = `""`;
